### PR TITLE
Snap: When snapping a directory, look for snap.yaml.

### PR DESCRIPTION
**Pull request by kyrofa**
Fixes LP: [#1541987](https://bugs.launchpad.net/snapcraft/+bug/1541987)

### CODE REVIEW

**Comment by kyrofa** [comment]
Adding an integration test here, don't approve just yet.

File: integration_tests/test_snap.py [hunk 1]
```diff
@@ -83,6 +83,19 @@ def test_snap(self):
             os.path.join('snap', 'bin', 'not-wrapped.wrapper'),
             Not(FileExists()))
 
+    def test_snap_directory(self):
+        project_dir = 'assemble'
+        self.run_snapcraft('snap', project_dir)
+        os.chdir(project_dir)
+
+        snap_file_path = 'assemble_1.0_{}.snap'.format(_get_deb_arch())
+        self.assertThat(snap_file_path, FileExists())
```
Comment (sergiusens) [inline integration_tests/test_snap.py]: maybe delete the snap here

File: integration_tests/test_snap.py [hunk 2]
```diff
@@ -83,6 +83,19 @@ def test_snap(self):
...
+
+        # Verify that Snapcraft can snap its own snap directory (this will make
+        # sure `snapcraft snap` and `snapcraft snap <directory>` are always in
+        # sync).
+        self.run_snapcraft(['snap', 'snap'])
```
Comment (sergiusens) [inline integration_tests/test_snap.py]: and check for existence after this

Comment (kyrofa) [inline integration_tests/test_snap.py]: Ah, good ideas. Done.

**Comment by sergiusens** [comment]
:+1:

**Pull request merged**